ui/page/root/home_page.go

```diff
@@ -97,12 +98,12 @@ func NewHomePage(l *load.Load) *HomePage {
 	}
 
 	go func() {
-		hp.isConnected = libutils.IsOnline()
+		hp.isConnected.Store(libutils.IsOnline())
 	}()
 
 	// init shared page functions
 	toggleSync := func(unlock load.NeedUnlockRestore) {
-		if !hp.WL.SelectedWallet.Wallet.IsConnectedToNetwork() {
+		if hp.WL.SelectedWallet.Wallet.IsConnectedToNetwork() {
```

Comment:
This is the only change needed... but please have a look at the rest as I think it was missing some thread safety for the isConnected bool and a way out of the connecting loop if ctx is canceled.